tools/explorer/lib/ontology.py

```diff
@@ -31,35 +32,137 @@ def GetFieldsForTypeName(self,
                            namespace: str,
                            entity_type_name: str,
                            required_only: bool = False) -> List[ETF]:
-    """Gets a list of fields for a given typename within a namespace.
+    """Gets a list of fields for a given typename within a namespace
+
+       Args:
+            namespace: the name of the namespace as a string
+            entity_type_name: name of the entity type as a string
+            required_only: when true will return
+            only required fields for a given type
+
+       Returns:
+            result_fields: a list of StandardField tuples
+    """
+    entity_type = self.universe.entity_type_universe.GetEntityType(
+        namespace,
+        entity_type_name
+    )
+    qualified_fields = entity_type.GetAllFields()
+    field_optwrappers = [
+        entity_type.GetField(field)
+        for field in qualified_fields
+    ]
+    entity_type_fields = [
+        ETF(namespace,
+            field.field.field[0:],
+            not field.optional,
+            field.field.increment) for field in field_optwrappers
+    ]
+    if required_only:
+      fields_temp = entity_type_fields
+      entity_type_fields = []
+      for field in fields_temp:
+        if field.IsRequired():
+          entity_type_fields.append(field)
+
+    return entity_type_fields
+
+  #TODO: Expand NONE matching logic
+  def _CreateMatch(
+      self,
+      field_list: List[StandardField],
+      entity_type: EntityType
+  ) -> str:
+    """Determines the closeness of the match between
+    the field list and the entity type
 
     Args:
-       namespace: the name of the namespace as a string.
-       entity_type_name: the name of the entity type as a string.
-       required_only: when true will return only required fields for a given
-         type.
+      field_list: a list of StandardField objects
+      entity_type: An EntityType object
 
     Returns:
-            result_fields: a list of StandardField tuples
+      The match type as a string
     """
-    pass
+    match_type = ''
+
+    entity_field_optwrappers = entity_type.GetAllFields().values()
+    entity_field_set = {
+        StandardField(entity_type.namespace.namespace, ow.field.field)
+        for ow in entity_field_optwrappers
+    }
+    input_field_set = set(field_list)
+
+    if entity_field_set.issubset(input_field_set):
+      if input_field_set.issubset(entity_field_set):
+        match_type = 'EXACT'
```

Comment:
Given that you're using sets of standard fields, it's possible for a non-exact match to show as exact in the case where one or both entities have incremented fields (run_status_1, run_status_2).  Is this as intended?

Reply:
Changing these two EntityTypeFields will fix this.